warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExpandCompositeTerms.java

```diff
@@ -332,6 +403,7 @@ public Object visit(ASTReferenceExpression node, Object data) {
         List<JexlNode> unmodifiedNodes = new ArrayList<>();
         List<JexlNode> modifiedNodes = new ArrayList<>();
         for (JexlNode nonLeafNode : nonLeafNodes) {
+            log.debug("Processing non leaf node {}", JexlStringBuildingVisitor.buildQuery(nonLeafNode));
```

Comment:
I wonder if this is worth guarding with `log.isDebugEnabled()`, since `JexlStringBuildingVisitor.buildQuery(nonLeafNode)` will still be evaluated no matter what. The placeholder just prevents the string concatenation in this case.

```suggestion
            if (log.isDebugEnabled()) {
                log.debug("Processing non leaf node {}", JexlStringBuildingVisitor.buildQuery(nonLeafNode));
            }
```

Reply:
Yeah that's a good point. It's a single node, but it's better to avoid that cost anyway.